numba/types.py

```diff
@@ -820,6 +820,9 @@ def strides(self):
              stride *= i
         return tuple(reversed(strides))
 
+    @property
+    def key(self):
+         return self.dtype, self.shape
```

Comment:
Sorry for nitpicking, but this line is slightly misindented :-) Otherwise, patch looks good to me.